### PR TITLE
Backlog 34960

### PR DESCRIPTION
**Pull request by patitapaban19**
Adding Azure Sql Database, SQL data sources types to PDI.
We have not added support for Azure Active Directory - Integrated authentication mode as we have an open issue going on with Microsoft regarding that.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 43m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **5065**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 23m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
:ok_hand: All tests passed!
 
Tests run: **5053**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 23m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
:ok_hand: All tests passed!
 
Tests run: **5053**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 24m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.step.mqtt.MQTTStreamSourceTest.testMqttStreamSingleTopic</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **5053**, Failures: **1**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
:ok_hand: All tests passed!
 
Tests run: **5061**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 24m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.step.mqtt.MQTTStreamSourceTest.testMqttStreamSingleTopic</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **5061**, Failures: **1**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/test/java/org/pentaho/di/core/database/AzureSqlDataBaseMetaTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+package org.pentaho.di.core.database;
```
Comment (lucboudreau) [inline core/src/test/java/org/pentaho/di/core/database/AzureSqlDataBaseMetaTest.java:1]: This file needs a license header.

**Review by lucboudreau — CHANGES_REQUESTED** [review]
Did a second read through. One file is missing a license header. Please add.

**Comment by buildguy** [comment]
### :warning: Build finished in 23m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 18 test(s) failed:

<details><summary><b>org.pentaho.di.trans.StepWithMappingMetaTest.loadMappingMeta</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.StepWithMappingMetaTest.loadMappingMetaTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.StepWithMappingMetaTest.testFileNameAsVariable</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.TransMetaTest.testLoadXml</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.TransMetaTest.testSetInternalEntryCurrentDirectoryWithRepository</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.TransMetaTest.testTransWithOneStepIsConsideredUsed</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.TransTest.testLoggingObjectIsNotLeakInTrans</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestCantFindDirectory</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestJobNotFoundInDirectory</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestCantFindRepository</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteJobServletTest.testExecuteJobServletTestWithUnauthorizedUser</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteTransServletTest.doGetRepositoryAuthenticationFailTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteTransServletTest.doGetTransformationNotFoundTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteTransServletTest.doGetTransformationFailsTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.www.ExecuteTransServletTest.doGetTransformationSucceedsTest</b> (click to expand)</summary><p>

```
javassist.CannotCompileException: by java.util.zip.ZipException: invalid distance too far back
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testJsonInputExternalResourceConsumer</b> (click to expand)</summary><p>

```
org/pentaho/platform/engine/core/system/PentahoSystem
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testJsonOutputExternalResourceConsumer</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.metaverse.api.analyzer.kettle.KettleAnalyzerUtil
```

</p></details>

  
Tests run: **5061**, Failures: **18**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 23m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog
```
---
:ok_hand: All tests passed!
 
Tests run: **5061**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7877/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**